nixos/modules/services/hardware/bluetooth.nix

```diff
@@ -28,18 +31,25 @@ in
     environment.systemPackages = [ bluez-bluetooth pkgs.openobex pkgs.obexftp ];
 
     services.udev.packages = [ bluez-bluetooth ];
-
     services.dbus.packages = [ bluez-bluetooth ];
+    systemd.packages       = [ bluez-bluetooth ];
+
+    services.udev.extraRules = optionalString config.hardware.bluetooth.powerOnBoot ''
+      ACTION=="add", KERNEL=="hci0", RUN+="${pkgs.bluez.out}/bin/hciconfig %k up"
```

Comment:
Is there a way to match bluetooth devices rather then relying on hard-coded interface names?